hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/Iterators.scala

```diff
@@ -346,75 +346,43 @@ object LogFileIterator extends SparkAdapterSupport {
-        .withStorage(storage)
-        .withBasePath(tablePath)
-        .withLogFilePaths(logFiles.map(logFile => logFile.getPath.toString).asJava)
-        .withReaderSchema(logSchema)
-        // NOTE: This part shall only be reached when at least one log is present in the file-group
-        //       entailing that table has to have at least one commit
-        .withLatestInstantTime(tableState.latestCommitTimestamp.get)
-        .withReverseReader(false)
-        .withInternalSchema(internalSchema)
-        .withBufferSize(
-          hadoopConf.getInt(HoodieMemoryConfig.MAX_DFS_STREAM_BUFFER_SIZE.key(),
-            HoodieMemoryConfig.DEFAULT_MR_MAX_DFS_STREAM_BUFFER_SIZE))
-        .withMaxMemorySizeInBytes(maxCompactionMemoryInBytes)
-        .withSpillableMapBasePath(
-          hadoopConf.get(HoodieMemoryConfig.SPILLABLE_MAP_BASE_PATH.key,
-            FileIOUtils.getDefaultSpillableMapBasePath))
-        .withDiskMapType(
-          hadoopConf.getEnum(HoodieCommonConfig.SPILLABLE_DISK_MAP_TYPE.key,
-            HoodieCommonConfig.SPILLABLE_DISK_MAP_TYPE.defaultValue))
-        .withBitCaskDiskMapCompressionEnabled(
-          hadoopConf.getBoolean(HoodieCommonConfig.DISK_MAP_BITCASK_COMPRESSION_ENABLED.key(),
-            HoodieCommonConfig.DISK_MAP_BITCASK_COMPRESSION_ENABLED.defaultValue()))
-
-      if (logFiles.nonEmpty) {
-        logRecordScannerBuilder.withPartition(getRelativePartitionPath(
-          new StoragePath(tableState.tablePath), logFiles.head.getPath.getParent))
-      }
+      .withDiskMapType(
+        hadoopConf.getEnum(HoodieCommonConfig.SPILLABLE_DISK_MAP_TYPE.key,
+          HoodieCommonConfig.SPILLABLE_DISK_MAP_TYPE.defaultValue))
+      .withBitCaskDiskMapCompressionEnabled(
+        hadoopConf.getBoolean(HoodieCommonConfig.DISK_MAP_BITCASK_COMPRESSION_ENABLED.key(),
+          HoodieCommonConfig.DISK_MAP_BITCASK_COMPRESSION_ENABLED.defaultValue()))
+
+    if (logFiles.nonEmpty) {
+      logRecordScannerBuilder.withPartition(getRelativePartitionPath(
+        new StoragePath(tableState.tablePath), logFiles.head.getPath.getParent))
+    }
 
-      logRecordScannerBuilder.withRecordMerger(
-        HoodieRecordUtils.createRecordMerger(tableState.tablePath, EngineType.SPARK, tableState.recordMergeImplClasses.asJava, tableState.recordMergeStrategyId))
+    logRecordScannerBuilder.withRecordMerger(
+      HoodieRecordUtils.createRecordMerger(tableState.tablePath, EngineType.SPARK, tableState.recordMergeImplClasses.asJava, tableState.recordMergeStrategyId))
 
-      val scanner = logRecordScannerBuilder.build()
+    val scanner = logRecordScannerBuilder.build()
 
-      closing(scanner) {
-        // NOTE: We have to copy record-map (by default immutable copy is exposed)
-        mutable.HashMap(scanner.getRecords.asScala.toSeq: _*)
-      }
+    closing(scanner) {
+      // NOTE: We have to copy record-map (by default immutable copy is exposed)
+      mutable.HashMap(scanner.getRecords.asScala.toSeq: _*)
```

Comment:
not related to this change but the spillable map is copied into a in-memory hashmap?

Reply:
Looks like it, we should be able to get rid of this reader path soon